myia/operations/prim_scalar_cast.py

```diff
@@ -34,7 +34,8 @@ def pyimpl_scalar_cast(x, t):
     if not isinstance(a, AbstractScalar):
         raise MyiaTypeError('scalar_cast must cast to a scalar type')
     t = a.xtype()
-    engine.check(xtype.Number, t)
+    # Let's accept both numbers and boolean.
+    engine.check((xtype.Number, xtype.Bool), t)
```

Comment:
Casting a number to bool is currently implemented as `x != 0` and I'm not sure we want two ways of doing it. Also, this allows int->bool, but bool->int will not work (because scalar is still required to only have type Number).

Reply:
Ok, I will remove that !